docs/tutorials/security.md

```diff
@@ -0,0 +1,74 @@
+  - Place an encrypted order, and reveal/execute it later time, or use the next feature of Tezos named TimeLock. The goal is the same: hide the transaction until the last moment no one can place a previous order. This requires a smart contract to handle this kind of order and keep track of previous data to be able to execute the transaction. Ex: Anterior currency pairs history values in the context of a swap for example.
+
+### Onchain attacks
+
+This training session will cover the topic of On-chain attacks, which are cyberattacks that target the blockchain network and its components. On-chain attacks can exploit various types of vulnerabilities, such as:
+
+- Programming errors that affect the security and functionality of smart contracts, which are self-executing agreements that run on the blockchain. Smart contracts can have a larger attack surface than traditional applications, as they interact with other contracts and users on the network and so, are subject to higher interest from hackers in draining funds.
+- Different kinds of leaks :
+  - Replay attacks: These occur when an attacker intercepts and retransmits a valid transaction on a different blockchain network, causing the same transaction to be executed twice
+  - Memory overflow: This happens when a smart contract runs out of memory or enters an unexpected context to exploit a part of the code.
+  - Re-entrancy attacks: These are a type of recursive call vulnerability that allows an attacker to repeatedly call a function within a smart contract before the previous call is finished, resulting in multiple withdrawals or transfers of funds
+- Blockchain user trust and management :
+  - Administrators: These are the entities that have the highest level of authority and control over the contract. They can perform actions such as deploying, upgrading, pausing, or terminating the contract
+  - Lambda / mutable code: This refers to the parts of the contract code that can be changed after deployment, such as parameters, variables, or functions.
+  - Check roles: This is the process of verifying whether a user has the right to execute a certain part of the code, such as calling a function or modifying a state variable.
+  - Trustable oracles: These are the external sources of data that your contract relies on, such as prices, events, or outcomes.
+
+In this training session, we will use a hands-on approach to learn how to identify, prevent, and mitigate On-chain attacks. We will work with a sample code that contains several bugs and vulnerabilities, and we will try to fix them step by step. We will also discuss best practices and recommendations for developing secure and reliable smart contracts and blockchain applications.
+
+## Prerequisites
+
+To run the code, a Ligo compiler is required and can be installed at this [location](https://ligolang.org/docs/intro/installation/?lang=jsligo)
+
+Compile the contract with the Ligo compiler
```

Comment:
This is a little confusing because I don't have a contract to compile yet so I don't know what to do with these commands.

Reply:
yep, it could be a bit confusing